File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        # min heap of size k
        minHeap = []

        for x, y in points:
            distance = (x**2) + (y**2)
            minHeap.append([distance, x, y])

        heapq.heapify(minHeap)

        res = []

        while k > 0:
            dist, x, y = heapq.heappop(minHeap)
            res.append([x, y])
            k -= 1
        return res

File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.cpp
class Solution
{
public:
  vector<vector<int>> kClosest(vector<vector<int>> &points, int k)
  {
    priority_queue<vector<int>> pq{};
    vector<vector<int>> res(k);
    for (auto &&point : points)
    {
      int x = point[0];
      int y = point[1];
      int dist = calculate(x, y);
      pq.push({dist, x, y});
      if (pq.size() > k)
      {
        pq.pop();
      }
    }
    for (int i = 0; i < k; i++)
    {
      vector<int> top = pq.top();
      pq.pop();
      res[i] = {top[1], top[2]};
    }
    return res;
  }
  int calculate(int x, int y)
  {
    return x * x + y * y;
  }
};